app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,93 @@
+require './config/environment'
+
+class TweetController < ApplicationController
+
+  get '/tweets' do
+    if session[:user_id]
```

Comment:
You can use `if current_user` here. Because in the lifespan of a request, you might call `session[:user_id]` multiple times, we'd prefer to only hit the database once via a call to `current_user` 

This pattern of conditionally setting an instance variable or returning it if it's already defined using the `||=` (or-equals) operator as you did in your `current_user` method is called Memoization ([read more](http://gavinmiller.io/2013/basics-of-ruby-memoization/) and [here](http://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/)).